### PR TITLE
Feat: work with the store instead of local states

### PR DESCRIPTION
**Pull request by ErezBiren**

### CODE REVIEW

File: src/pages/NewQuestionPage.tsx [hunk 1]
```diff
@@ -1,45 +1,61 @@
...
+  const navigate = useNavigate();
 
-  const handleChange = (newTags: MuiChipsInputChip[]) => {
-    setTags(newTags);
+  const addQuestion = useStore((state: StoreState) => state.addQuestion);
+
+  const [question, setQuestion] = useState<Question>({
+    ...initialQuestionObject,
+    id: crypto.randomUUID(),
+  });
```
Comment (AmitGit217) [inline src/pages/NewQuestionPage.tsx:16]: Ok I gave you in the interface the ID column I mislead you a bit sorry. This column will be autogenerated in the back-end

File: src/pages/NewQuestionPage.tsx [hunk 2]
```diff
@@ -1,45 +1,61 @@
...
-function NewQuestion() {
-  const [title, setTitle] = useState("");
-  const [tags, setTags] = useState<MuiChipsInputChip[]>(["aaaaa","bbbbbb"]);
-  const [text, setText] = useState("");
+const NewQuestion = () => {
+  const navigate = useNavigate();
 
-  const handleChange = (newTags: MuiChipsInputChip[]) => {
-    setTags(newTags);
+  const addQuestion = useStore((state: StoreState) => state.addQuestion);
```
Comment (AmitGit217) [inline src/pages/NewQuestionPage.tsx:11]: Great!

File: src/store/store.ts [hunk 3]
```diff
@@ -2,7 +2,9 @@ import { create } from "zustand";
 import { AuthSlice, createAuthSlice } from "./auth-slice";
 import { QuestionsSlice, createQuestionsSlice } from "./questions-slice";
 
-const useStore = create<AuthSlice & QuestionsSlice>()((...a) => ({
+export type StoreState = AuthSlice & QuestionsSlice;
+
+const useStore = create<StoreState>()((...a) => ({
   ...createQuestionsSlice(...a),
   ...createAuthSlice(...a),
 }));
```
Comment (AmitGit217) [inline src/store/store.ts:10]: 1. `(...a)` Can you make it a bit more informative? I know it's only for the store but still, it's all about habits.

**Review by AmitGit217 — APPROVED** [review]

File: src/store/store.ts (hunk 3 above)
Comment (ErezBiren) [reply src/store/store.ts:10]: You're absolutely right!
It bothered me also to leave it like that, I just didn't find yet what should I call it.

**Pull request merged**